pywatts/modules/preprocessing/select.py

```diff
@@ -0,0 +1,92 @@
+from typing import Dict, List, Union
+
+import xarray as xr
+from pywatts_pipeline.core.transformer.base import BaseTransformer
+from pywatts_pipeline.utils._xarray_time_series_utils import _get_time_indexes
+
+
+class Select(BaseTransformer):
+    """
+    This module selects for each time stamp one or multiple values from the original time series.
+    E.g. it can be used for building input samples, target samples, or for shifting the time series.
+
+    :param start: The offset for shifting the time series. If this is a list, stop and step are ignored and
+                  the list specifies the values that should be selected.
+    :type start: Union[int, List[int]
+    :param stop: The offset for shifting the time series
+    :type stop: int
+    :param step: The offset for shifting the time series
+    :type step: int
+    """
+
+    def __init__(
+        self,
+        start: Union[int, List[int]],
+        stop: int = None,
+        step: int = None,
+        name: str = "SampleModule",
+    ):
+        super().__init__(name)
+        self.start = start
+        self.stop = stop
+        self.step = step
+
+    def get_min_data(self):
+        if isinstance(self.start, list):
+            return abs(min(self.start))
+        return abs(self.start) if self.start < 0 else 0
+
+    def get_params(self) -> Dict[str, object]:
+        """
+        Returns a list of all hyperparameters/ user defined parameters
+
+        :return: Dict with params
+        :rtype: dict
+        """
+        return {
+            "start": self.start,
+            "stop": self.stop,
+            "step": self.step,
+        }
+
+    def set_params(
+        self, start: Union[int, List[int]] = None, stop: int = None, step: int = None, **kwargs
+    ):
+        """
+        Set params.
+
+
+        :param start: The offset for shifting the time series. If this is a list, stop and step are ignored and
+                      the list specifies the values that should be selected.
+        :type start: Union[int, List[int]
+        :param stop: The offset for shifting the time series
+        :type stop: int
+        :param step: The offset for shifting the time series
+        :type step: int
+        """
+        if start:
+            self.start = start
+        if stop:
+            self.stop = stop
+        if step:
+            self.step = step
+
+    def transform(self, x: xr.DataArray) -> xr.DataArray:
+        """
+        Select the desired subset from the given time series x
+
+        :param x: the input
+        :type x: xr.DataArray
+        :return: A shifted time series.
+        :rtype: xr.DataArray
+        """
+        indexes = _get_time_indexes(x)
+        if isinstance(self.start, list):
```

Comment:
Why check if self.start ist list? In the tests start always is integer.

Reply:
Thank you, we should add a further test which passes a list. E.g. if a forecaster aims to consider the 1, 24 and 168 hour lags.